cluster/src/main/java/org/apache/iotdb/cluster/query/ClusterPlanRouter.java

```diff
@@ -292,4 +295,64 @@ private PartitionGroup routePlan(ShowChildPathsPlan plan) {
     }
     return result;
   }
+
+  @SuppressWarnings("SuspiciousSystemArraycopy")
+  private Map<PhysicalPlan, PartitionGroup> splitAndRoutePlan(CreateMultiTimeSeriesPlan plan)
+    throws MetadataException {
+    Map<PhysicalPlan, PartitionGroup> result = new HashMap<>();
+    Map<PartitionGroup, PhysicalPlan> groupHoldPlan = new HashMap<>();
+
+    for (int i = 0; i < plan.getPaths().size(); i++) {
+      PartialPath path = plan.getPaths().get(i);
+      PartitionGroup partitionGroup =
+        partitionTable.partitionByPathTime(path, 0);
+      CreateMultiTimeSeriesPlan subPlan = null;
+      if (groupHoldPlan.get(partitionGroup) == null) {
```

Comment:
Assign `groupHoldPlan.get(partitionGroup)` to a local variable so that you will not need to call `get` twice.

Reply:
good